packages/datetime/src/timePicker.tsx

```diff
@@ -252,10 +274,30 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         this.updateTime(loopTime(newTime, unit), unit);
     }
 
+    private isTimeInRange(date: Date) {
+        const format = "H:mm:ss:SSS";
+        const time = moment(this.getDateOnlyWithTime(date), format);
+        const minTime = moment(this.getDateOnlyWithTime(this.props.minTime), format);
+        const maxTime = moment(this.getDateOnlyWithTime(this.props.maxTime), format);
+
+        const isTimeAfterMinTime = time.isSameOrAfter(minTime);
+        const isTimeBeforeMaxTime = time.isSameOrBefore(maxTime);
+
+        if (maxTime.isBefore(minTime)) {
+            return isTimeAfterMinTime || isTimeBeforeMaxTime;
+        }
+
+        return isTimeAfterMinTime && isTimeBeforeMaxTime;
+    }
+
+    private getDateOnlyWithTime(date: Date): Date {
```

Comment:
i think we have a Util function for this? or, we should; this doesn't belong as instance method. @cmslewis ?